server/cluster/shard.go

```diff
@@ -28,33 +31,148 @@ type ShardTablesWithRole struct {
 // Shard FSM Const Definition
 // It contains the event name and the state name
 const (
-	EventTransferLeader          = "TransferLeader"
-	EventTransferFollowerFailed  = "TransferFollowerFailed"
-	EventTransferFollower        = "TransferFollower"
-	EventTransferLeaderFailed    = "TransferLeaderFailed"
+	StateLeader          = "LEADER"
```

Comment:
New a sub package in `coordinator/procedure` like `role` or `shard`. Split leader and follower fsm.

Reply:
fix